src/main/groovy/io/jenkins/infra/repository_permissions_updater/ArtifactoryPermissionsUpdater.groovy

```diff
@@ -146,18 +146,42 @@ class ArtifactoryPermissionsUpdater {
                         users [:]
                     } else {
                         users definition.developers.collectEntries { developer ->
-                            if (!KnownUsers.exists(developer.toLowerCase())) {
-                                reportChecksApiDetails(developer + " needs to login to artifactory", 
+                            def existsInArtifactory = KnownUsers.existsInArtifactory(developer)
+                            def existsInJira = KnownUsers.existsInJira(developer) || JiraAPI.getInstance().isUserPresent(developer)
+
+                            if (!existsInArtifactory && !existsInJira) {
+                                reportChecksApiDetails(developer + " needs to log in to Artifactory",
                                 """
-                                ${developer} needs to login to [artifactory](https://repo.jenkins-ci.org/).
+                                ${developer} needs to log in to [Artifactory](https://repo.jenkins-ci.org/) and [Jira](https://issues.jenkins.io/).
 
-                                We resync our artifactory list hourly, so you will need to wait some time before rebuilding your pull request.
+                                We resync our Artifactory list hourly, so you will need to wait some time before rebuilding your pull request.
+                                The easiest way to trigger a rebuild is to close your pull request, wait a few seconds and then reopen it.
+
+                                Alternatively the hosting team can re-trigger it if you post a comment saying you have now logged in.
+                                """.stripIndent())
+                                throw new IllegalStateException("User name not known to Artifactory and Jira: " + developer)
+                            }
+
+                            if (!existsInArtifactory) {
+                                reportChecksApiDetails(developer + " needs to log in to Artifactory",
+                                        """
+                                ${developer} needs to log in to [Artifactory](https://repo.jenkins-ci.org/).
+
+                                We resync our Artifactory list hourly, so you will need to wait some time before rebuilding your pull request.
                                 The easiest way to trigger a rebuild is to close your pull request, wait a few seconds and then reopen it.
 
                                 Alternatively the hosting team can re-trigger it if you post a comment saying you have now logged in.
                                 """.stripIndent())
                                 throw new IllegalStateException("User name not known to Artifactory: " + developer)
                             }
+
+                            if (!existsInJira) {
```

Comment:
shouldn't we re-work this to check both jira and artifactory before throwing an exception and also produce a  different checks report if both are missing?

Reply:
https://github.com/jenkins-infra/repository-permissions-updater/pull/1856/files#diff-3e4cb1ddb9f0646c5ddc1302282ba93e58eb6d012fab254ecd90cad8b7202283R155

Reply:
ah i missed the first check